src/Microsoft.AspNet.Routing.Extensions/RequestDelegateRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,109 @@
+
+namespace Microsoft.AspNet.Builder
+{
+    public static class RequestDelegateRouteBuilderExtensions
+    {
+        public static IRouteBuilder MapRoute(this IRouteBuilder builder, string template, RequestDelegate handler)
+        {
+            var route = new Route(
+                new RouteHandler(handler),
+                template,
+                defaults: null,
+                constraints: null,
+                dataTokens: null,
+                inlineConstraintResolver: GetConstraintResolver(builder));
+
+            builder.Routes.Add(route);
+            return builder;
+        }
+
+        public static IRouteBuilder MapRoute(this IRouteBuilder builder, string template, Action<IApplicationBuilder> action)
+        {
+            var nested = builder.ApplicationBuilder.New();
+            action(nested);
+            return builder.MapRoute(template, nested.Build());
+        }
+
+        public static IRouteBuilder MapDelete(this IRouteBuilder builder, string template, RequestDelegate handler)
+        {
+            return builder.MapVerb("DELETE", template, handler);
+        }
+
+        public static IRouteBuilder MapDelete(this IRouteBuilder builder, string template, Action<IApplicationBuilder> action)
+        {
+            return builder.MapVerb("DELETE", template, action);
+        }
+
+        public static IRouteBuilder MapGet(this IRouteBuilder builder, string template, RequestDelegate handler)
+        {
+            return builder.MapVerb("GET", template, handler);
+        }
+
+        public static IRouteBuilder MapGet(this IRouteBuilder builder, string template, Action<IApplicationBuilder> action)
+        {
+            return builder.MapVerb("GET", template, action);
+        }
+
+        public static IRouteBuilder MapPost(this IRouteBuilder builder, string template, RequestDelegate handler)
+        {
+            return builder.MapVerb("POST", template, handler);
+        }
+
+        public static IRouteBuilder MapPost(this IRouteBuilder builder, string template, Action<IApplicationBuilder> action)
+        {
+            return builder.MapVerb("POST", template, action);
+        }
+
+        public static IRouteBuilder MapPut(this IRouteBuilder builder, string template, RequestDelegate handler)
+        {
+            return builder.MapVerb("PUT", template, handler);
+        }
+
+        public static IRouteBuilder MapPut(this IRouteBuilder builder, string template, Action<IApplicationBuilder> action)
+        {
+            return builder.MapVerb("PUT", template, action);
+        }
+
+        public static IRouteBuilder MapVerb(
+            this IRouteBuilder builder,
+            string verb,
+            string template,
+            Action<IApplicationBuilder> action)
+        {
+            var nested = builder.ApplicationBuilder.New();
+            action(nested);
+            return builder.MapVerb(verb, template, nested.Build());
+        }
+
+        public static IRouteBuilder MapVerb(
```

Comment:
I'd rather wait for user demand. All of this stuff can be written pretty easily.